### PR TITLE
Refactor by introducing a new `PlayerID` class

### PR DESCRIPTION
**Pull request by Yo1k**

### CODE REVIEW

File: yo1k/tic_tac_toe/kernel/game.py [hunk 1]
```diff
@@ -11,15 +11,27 @@ class Mark(Enum):
...
+
+    def __repr__(self) -> str:
+        return (f"{type(self).__qualname__}("
+                f"idx={self.idx})")
+
+
 @eq
 class Player:
-    def __init__(self, mark: Mark):
+    def __init__(self, mark: Mark, id_: PlayerID):
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/kernel/game.py]: I already have `id` as the first constructor parameter. Let's maintain the same order in Python (this is also relevant for the `_new_player` function in tests).

File: yo1k/tic_tac_toe/kernel/game.py [hunk 2]
```diff
@@ -96,25 +108,28 @@ def __init__(
...
-        self.required_ready: set[int] \
-            = set(range(len(players))) if required_ready is None else required_ready
+        self.required_ready: set[PlayerID] \
+            = set(player.id for player in players) if required_ready is None \
+            else required_ready
 
-    def turn(self) -> int:
-        """Returns `turn` - the index of the current `Player` in `State.players`.
+    def turn(self) -> PlayerID:
+        """Returns `turn` - the id of the current `Player` in the `State.players`.
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/kernel/game.py]: We should say either _"the ID"_ or _"the `id`"_, but not _"the id"_.

File: yo1k/tic_tac_toe/kernel/game.py [hunk 3]
```diff
@@ -96,25 +108,28 @@ def __init__(
...
 
-        `turn` is calculated in such a way as to alternate players order in different
-        `round` while keeping the same `Player` index in the `State.players`.
+        `turn` is calculated in such a way as to alternate players order
+         in the different `round`.
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/kernel/game.py]: "the" is not needed here. Let's just say "to alternate players order in different rounds".

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: yo1k/tic_tac_toe/kernel/game.py (hunk 1 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/kernel/game.py]: Done.

File: yo1k/tic_tac_toe/kernel/game.py (hunk 2 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/kernel/game.py]: Done.

File: yo1k/tic_tac_toe/kernel/game.py (hunk 3 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/kernel/game.py]: Done.

**Review by stIncMale — APPROVED** [review]

**Pull request merged**